test/unit/test_data_utils.py

```diff
@@ -14,8 +14,8 @@
 import unittest
 import os
 from pathlib import Path
+import shutil
 import signal
```

Comment:
You can remove this, since you also removed the only instance it was used.

Reply:
It is still being used. I deleted the `_clear_folder` function and started using `shutil.rmtree` in `_check_piped_dmatrix`.